packages/fdc3-agent-proxy/src/BasicDesktopAgent.ts

```diff
@@ -13,36 +13,36 @@ import {
 import { ChannelSupport } from './channels/ChannelSupport';
 import { AppSupport } from './apps/AppSupport';
 import { IntentSupport } from './intents/IntentSupport';
-import { HandshakeSupport } from './handshake/HandshakeSupport';
 import { Connectable } from '@kite9/fdc3-standard';
 import { Context } from '@kite9/fdc3-context';
+import { HeartbeatSupport } from './heartbeat/HeartbeatSupport';
 
 /**
  * This splits out the functionality of the desktop agent into
  * app, channels and intents concerns.
  */
 export class BasicDesktopAgent implements DesktopAgent, Connectable {
-  readonly handshake: HandshakeSupport;
+  readonly heartbeat: HeartbeatSupport;
   readonly channels: ChannelSupport;
   readonly intents: IntentSupport;
   readonly apps: AppSupport;
   readonly connectables: Connectable[];
 
   constructor(
-    handshake: HandshakeSupport,
+    heartbeat: HeartbeatSupport,
     channels: ChannelSupport,
     intents: IntentSupport,
     apps: AppSupport,
     connectables: Connectable[]
   ) {
-    this.handshake = handshake;
+    this.heartbeat = heartbeat;
     this.intents = intents;
     this.channels = channels;
     this.apps = apps;
     this.connectables = connectables;
   }
 
-  async addEventListener(type: FDC3EventTypes | null, handler: EventHandler): Promise<Listener> {
+  addEventListener(type: FDC3EventTypes | null, handler: EventHandler): Promise<Listener> {
```

Comment:
If this function is no longer `async`, should it still be marked as returning a `Promise`? Or are all of the returned values promises?

Reply:
Yes its passing through the promise from `DefaultChannelSupport.addChannelChangedEventHandler`.

All returned values are indeed promises - although the throw on line 50 should probably be a `Promise.reject` now, although I doubt it would make much difference.